src/lib.rs

```diff
@@ -33,6 +37,51 @@ pub fn links(bytes: &[u8], links: &mut impl Extend<Cid>) -> Result<(), Error> {
     Ok(())
 }
 
+/// DAG-PB implementation of ipld-core's `Codec` trait.
+pub struct DagPbCodec;
+
+impl Codec<Ipld> for DagPbCodec {
+    const CODE: u64 = 0x70;
+    type Error = Error;
+
+    fn decode<R: BufRead>(mut reader: R) -> Result<Ipld, Self::Error> {
+        let mut bytes = Vec::new();
+        reader.read_to_end(&mut bytes)?;
+        crate::to_ipld(&bytes)
+    }
+
+    fn encode<W: Write>(mut writer: W, data: &Ipld) -> Result<(), Self::Error> {
+        let bytes = crate::from_ipld(data)?;
+        Ok(writer.write_all(&bytes)?)
+    }
+}
+
+impl Links for DagPbCodec {
+    type LinksError = Error;
+
+    fn links(data: &[u8]) -> Result<impl Iterator<Item = Cid>, Self::LinksError> {
+        let mut links = Vec::new();
+        crate::links(data, &mut links)?;
+        Ok(links.into_iter())
+    }
+}
+
+impl Codec<PbNode> for DagPbCodec {
+    const CODE: u64 = 0x70;
+    type Error = Error;
+
+    fn decode<R: BufRead>(mut reader: R) -> Result<PbNode, Self::Error> {
+        let mut bytes = Vec::new();
+        reader.read_to_end(&mut bytes)?;
```

Comment:
We're converting from a reader, to bytes, then back to a reader in:

https://github.com/ipld/rust-ipld-dagpb/blob/f9dfacad8424dc10a1532c5c94107c06bace80e7/src/codec.rs#L42-L43

Reply:
Yes, bit it can't be changed easily due to the way the `BytesReader` works.